pkg/apis/kops/cluster.go

```diff
@@ -317,8 +317,12 @@ type ClusterSubnetSpec struct {
 	CIDR string `json:"cidr,omitempty"`
 
 	// ProviderID is the cloud provider id for the objects associated with the zone (the subnet on AWS)
-	ProviderID string `json:"id,omitempty"`
+	ProviderID string `json:"subnetId,omitempty"`
 
+	NgwId  string `json:"ngwId,omitempty"`
+	NgwEip string `json:"ngwEip,omitempty"`
+	// TODO: ACTIVATE EGRESSIP AND WRITE THE LOGIC TO ALLOW SMARTER AND BROADER SPECING OF EXISTING COMPONENTS
+	// EgressIP   string     `json:"egressIP,omitempty"`
```

Comment:
I would be much happier with `EgressIP` - this is the API so mulligans are really expensive.